### DISCUSSION TITLE
iOS - AudioPlayer is not playing, no notification, no error

### DISCUSSION THREAD

**Discussion by miksansegundo** [mediagrid/capacitor-native-audio#11] [4. Bugs] [answered]
Hi there! I'd appreciate some help here.

I'm testing on iOS and getting the success callback triggered, then nothing happens. The audio doesn't play, and no notification or error is shown.

The code follows the creation as in example at https://github.com/mediagrid/capacitor-native-audio/blob/72f95bd11c556c384b0ced1ba5d55e3b489d2433/example/src/js/example.ts#L20

**Comment by wsamoht** [comment]
I tried going to `******` and it is behind some basic auth. The URL needs to be accessible. If you need to restrict access to the audio, then you'll want to use a signed URL.

**Reply by miksansegundo** [reply]
Do you mind removing the URL from this comment, please?

**Reply by wsamoht** [reply]
Done! And deleted the revision history.

**Comment by miksansegundo** [comment]
Thanks so much for catching that. 

I have removed the auth and tried again with the same results. The audio does not play, and no notification is shown on iOS. I'm testing in the XCode Simulator iOS 18 on several iPhone and iPad models.  

Here are the logs. 
<img width="369" alt="Screenshot 2567-10-22 at 12 05 15" src="https://github.com/user-attachments/assets/1d5e199f-b789-4b65-9cc7-cc5c72b3cdfa">


Here is the code:
```js
async connectedCallback() {
      const audioId = 'radio';

      await AudioPlayer.create({
        audioId,
        audioSource: 'https://URL/sound.mp3',
        friendlyTitle: 'Radio',
        useForNotification: true,
        isBackgroundMusic: false,
        loop: false,
      })
      .then( () => console.log('Success' ))
      .catch( ex => console.log( ex ) );
}
```

When the config is `useForNotification: false` and `isBackgroundMusic: true` the error is `"There was an issue creating the audio player [1]` from the try catch [here](https://github.com/mediagrid/capacitor-native-audio/blob/72f95bd11c556c384b0ced1ba5d55e3b489d2433/ios/Sources/AudioPlayerPlugin/AudioPlayerPlugin.swift#L90) with the error "An audio source with useForNotification = true must exist first." from [here](https://github.com/mediagrid/capacitor-native-audio/blob/72f95bd11c556c384b0ced1ba5d55e3b489d2433/ios/Sources/AudioPlayerPlugin/AudioPlayerPlugin.swift#L75)

**Reply by wsamoht** [reply]
Are you calling `initialize` and `play` after `create`?

**Comment by Jakob-SD** [comment]
I'm getting the same error. If I call `initialize` and `play` efter `create`, I get more errors. My implementation is

```javascript
async init(){
    const audioId = 'test';
    const useForNotification = false;
    
    await AudioPlayer.create({
      audioId: audioId,
      audioSource: 'https://file-examples.com/storage/feb05093336710053a32bc1/2017/11/file_example_MP3_5MG.mp3',
      friendlyTitle: 'friendlyTitle',
      useForNotification: useForNotification,
      artworkSource: '',
      isBackgroundMusic: false,
      loop: false,
  }).catch(ex => console.log(ex))


  //await AudioPlayer.initialize({ audioId: audioId }).catch(ex => console.log(ex));
  //AudioPlayer.play({ audioId: audioId }).catch(ex => console.log(ex))
  }
```

**Reply by wsamoht** [reply]
You need to initialize the first audio source with `useForNotification = true` as the error says :)

**Reply by Jakob-SD** [reply]
Haha, that makes sence. That part of the error was not shown on my end. I got the following but changing the notification setting did the trick

![CleanShot 2024-10-22 at 18 39 34@2x](https://github.com/user-attachments/assets/75d78114-c97a-4d6d-841b-3740b88dd17b)

I apologize for the confusion

**Reply by miksansegundo** [reply]
It's still not working for me. Which code is working for you?

**Comment by Jakob-SD** [comment, marked as answer]
This works for me on iOS when running through Xcode. I call the `init()` method from a play button.

```javascript
async init(){

    const audioId = 'test';
    const useForNotification = true;
    
    await AudioPlayer.create({
      audioId: audioId,
      audioSource: 'https://file-examples.com/storage/feb05093336710053a32bc1/2017/11/file_example_MP3_5MG.mp3,
      friendlyTitle: 'friendlyTitle',
      useForNotification: useForNotification,
      artworkSource: '',
      isBackgroundMusic: false,
      loop: false,
  }).catch(ex => console.log(ex))


  await AudioPlayer.initialize({ audioId: audioId }).catch(ex => console.log(ex));
  AudioPlayer.play({ audioId: audioId }).catch(ex => console.log(ex))
  }
```

**Reply by miksansegundo** [reply]
Thanks, that works!
